OctopusDSC/Tests/cOctopusServer.Tests.ps1

```diff
@@ -92,6 +93,56 @@ try
+                    $SecureMasterKey = ConvertTo-SecureString $MasterKey -AsPlainText -Force
+                    $MasterKeyCred = New-Object System.Management.Automation.PSCredential  ("notused", $SecureMasterKey)
+
+                    $haparams = @{
+                        Ensure = "Present";
+                        State = "Started";
+                        Name = "HANode";
+                        WebListenPrefix = "http://localhost:82";
+                        SqlDbConnectionString = "Server=(local);Database=Octopus;Trusted_Connection=True;";
+                        OctopusMasterKey = $MasterKeyCred;
+                        OctopusAdminCredential = $cred;
+                        ListenPort = 10935;
+                        AllowCollectionOfAnonymousUsageStatistics = $false;
+                        HomeDirectory = "C:\ChezOctopusSecondNode";
+                    }
+
+                    Set-TargetResource @haparams
+
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 8 -Exactly
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 1 -Exactly -ParameterFilter { ($arguments -join ' ') -eq 'create-instance --console --instance HANode --config \Octopus\OctopusServer-HANode.config --home C:\ChezOctopusSecondNode' }
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 1 -Exactly -ParameterFilter { ($arguments -join ' ') -eq "database --instance HANode --connectionstring $($haparams['SqlDbConnectionString']) --masterKey $MasterKey --grant NT AUTHORITY\SYSTEM" }
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 1 -Exactly -ParameterFilter { ($arguments -join ' ') -eq "configure --console --instance HANode --upgradeCheck True --upgradeCheckWithStatistics False --webForceSSL False --webListenPrefixes $($haparams['WebListenPrefix']) --commsListenPort 10935 --autoLoginEnabled False" }
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 1 -Exactly -ParameterFilter { ($arguments -join ' ') -eq 'service --console --instance HANode --stop' }
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 1 -Exactly -ParameterFilter { ($arguments -join ' ') -eq 'admin --console --instance HANode --username Admin --password S3cur3P4ssphraseHere!' }
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 1 -Exactly -ParameterFilter { ($arguments -join ' ') -eq 'license --console --instance HANode --free' }
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 1 -Exactly -ParameterFilter { ($arguments -join ' ') -eq 'service --console --instance HANode --install --reconfigure --stop' }
+                    Assert-MockCalled -CommandName 'Invoke-OctopusServerCommand' -Times 1 -Exactly -ParameterFilter { ($arguments -join ' ') -eq 'service --start --console --instance HANode' }
```

Comment:
This is awesome, though I'm sure we can make it _look_ less daunting at some point